main.go

```diff
@@ -159,23 +161,34 @@ func (self *PostgresLogParser) parseLogBuffer() (*PostgresLogLine, error) {
 	timestamp := parseTime(self.buffer)
 	user, database := parseUserAndDatabase(self.buffer)
 	logType, statementName := parseLogTypeWithStatementName(self.buffer)
-	value := parseValueFromBuffer(self.buffer)
-	// duration := parseDuration(self.buffer)
+	value, shardPartition, partitionlessQuery := parseValuesFromBuffer(self.buffer)
 
 	log := &PostgresLogLine{
-		Timestamp:     timestamp,
-		Username:      user,
-		Database:      database,
-		Duration:      duration,
-		LogType:       logType,
-		StatementName: statementName,
-		Value:         value,
+		Timestamp:          timestamp,
+		Username:           user,
+		Database:           database,
+		Duration:           duration,
+		LogType:            logType,
+		StatementName:      statementName,
+		Value:              value,
+		ShardPartition:     shardPartition,
+		PartitionlessQuery: partitionlessQuery,
 	}
 
 	self.buffer = ""
 	return log, nil
 }
 
+func parseShardFromValue(value string) string {
+	shardPartition := ""
+	if strings.Contains(value, "abacus") {
```

Comment:
Is there a way we can do this without baking abacus into the mix?

Reply:
ya we can. I thought that you said it would always be abacus. Maybe I misheard. I'll make a more generic case

Reply:
It is for MX, but this is an open source project. It would be nice to make this more generic.